mne/preprocessing/ica.py

```diff
@@ -2680,8 +2672,18 @@ def read_ica_eeglab(fname):
     ica.pre_whitener_ = np.ones((len(eeg.icachansind), 1))
     ica.pca_mean_ = np.zeros(len(eeg.icachansind))
 
-    ica.unmixing_matrix_ = eeg.icaweights
-    ica.pca_components_ = eeg.icasphere
-    ica.mixing_matrix_ = np.linalg.pinv(ica.pca_components_) @ eeg.icawinv
-
+    n_ch = len(ica.ch_names)
+    assert eeg.icaweights.shape == (n_components, n_ch)
```

Comment:
@larsoner  I cannot add this directly as a suggestion that you can commit because it is not on lines part of this commit, but we would have to change line 2667 from "ica.ch_names = info["ch_names"]" to something like "ica.ch_names = np.array(info["ch_names"])[eeg["icachansind"].astype(int)-1].tolist()". EEGLAB ICA can be computed on a subset of channels (indicated by the icachansind field) and when they are, your assert on line 2676 fails. With this change, it passes.